src/pages/app/posting_project_user/ProfilePage.js

```diff
@@ -121,40 +151,61 @@ const ProfilePage = (props) => {
     projectId: null,
     projectName: null,
   };
+  let profileData = {
+    orgId: null,
+    orgName: null,
+    location: null,
+    university: null,
+    category: null,
+    description: null,
+    email: null,
+    facebook: null,
+    phoneNumber: null,
+    vision: null,
+  };
 
   //Declare useDispatch hook to use;
   const dispatch = useDispatch();
-  //Load profileData from global store into 'selectedData' variable using useSelector() hook
-  const selectedData = useSelector(
+
+  //Load data from global store into variables variable using useSelector() hook
+  const selectedProject = useSelector(
+    (state) => state.profileReducer.projectData.props
+  );
+  const selectedProfile = useSelector(
     (state) => state.profileReducer.profileData.props
   );
+
   //Update the data from the 'selectedData' variable into the local variable
-  //If the data is not equal to undefined (this mean the data is fetched succesfully) then
-  //assign it into the local variable.
-  //else, execute the FetchData() function.
-  if (selectedData != undefined) {
-    profileData = selectedData;
-  } else FetchData();
-  console.log("This is ProfileData name", profileData.orgName); // For testing purpose
-  console.log("Profile data saved succesfully:", profileData); //For testing purpose
-  //Function to fetch data and dispatch it into the global store. I will move this is the profileAction.js later
+  //If the data is equal to undefined (not fetched succesfully) then execute the FetchData() function.
+  if (selectedProject != undefined) {
+    projectData = selectedProject;
+  } else FetchData("public-projects", `${testing_project_id}`);
+  console.log("Profile data saved succesfully:", profileData.orgName); //For testing purpose
+  //Similar for Profile Data
+  if (selectedProfile != undefined) {
+    profileData = selectedProfile;
+  } else FetchData("organization", `${testing_organization_id}`);
+  console.log("Profile data saved succesfully:", profileData.orgName); //For testing purpose
```

Comment:
**Statements**
The code works well. You make it work Long. That's nice. However, something is nice, doesn't mean it's effective.
There are some problems in the code that need re-consideration and fixing

**Problems**
**1. Putting plain if-else in component**
We should not put plain if else statements in the component like this (outside a function), since it will run every time the component gets re-render
Eg: An example is useState. Every time the value, which is initialized by useState, changes, the whole component gets re-render. Then those if-else will also be executed, in this case, we'll end up running the `FetchData` multiple times (That's maybe the reason why the data is read more than 2k times on FireStore)
**2. Redundant logic**
Since the `FetchData` sends the data to the reducer (global state), we can retrieve them like a pro. There's no need to consider an if else in this case to eliminate the value of `selectedProjects` and even to run the `FetchData` based on that elimination
**Solution**
**useEffect and useCallback**
When mastering the connection between the 2 hooks, you can execute the action and send the data to global state
After that, you can retrieve them and put them in the UI (Client-side rendering)
Another convention to mention is that we should not put the back-end code within the front-end code like this. It will cause confusion, make our code messy and vulnerable (in terms of security)

It'd be much better if we put the code to fetch the data in the action store.
So please use the effect and callback hooks, we can execute them, and then useSelector to retrieve them

Reply:
1. If-else actually is my last option because I cannot use useEffect and useCallback. And I also know about the re-rendering stuff, just that I have no other way. Also, the 2k reading on FireBase is because of when I was testing my code, not when I finish it.
2. As I don't use useEffect and useCallback, if there were not these if else, the web will read the data even before the Data is fetched, which will case problems. Well you sure know it, you taught me that.
Solution: The thing with useEffect and useCallback is if I use them I cannot use the <Link> component, somehow. If I didn't need to navigate between pages, I would have used useCallback and useEffect already.

Reply:
Replaced the old code with the new one. Will resolve this conversation